frappe/public/js/frappe/form/formatters.js

```diff
@@ -61,7 +61,13 @@ frappe.form.formatters = {
 	},
 	Currency: function (value, docfield, options, doc) {
 		var currency  = frappe.meta.get_field_currency(docfield, doc);
-		var precision = docfield.precision || cint(frappe.boot.sysdefaults.currency_precision) || 2;
+		var precision;
+
+		if (docfield.precision === 0) {
+			precision = docfield.precision;
+		} else {
+			precision = docfield.precision || cint(frappe.boot.sysdefaults.currency_precision) || 2;
+		}
```

Comment:
This condition isn't required
```suggestion
		
	    precision = docfield.precision || cint(frappe.boot.sysdefaults.currency_precision) || 2;
	
```

Reply:
@prssanna I don't think this condition will work, if the precision is set as 0 it will take  `frappe.boot.sysdefaults.currency_precision` or 2 as precision

Reply:
@deepeshgarg007 You're right, the condition is necessary